src/main/scala/esmeta/ir/Ref.scala

```diff
@@ -1,25 +1,29 @@
 package esmeta.ir
 
 import esmeta.ir.util.Parser
+import esmeta.lang.Syntax
 
 // IR references
-sealed trait Ref extends IRElem
+sealed trait Ref extends IRElem:
+  val langOpt: Option[Syntax]
 object Ref extends Parser.From(Parser.ref)
 
-case class Prop(ref: Ref, expr: Expr) extends Ref
+case class Prop(ref: Ref, expr: Expr, langOpt: Option[Syntax] = None)
+  extends Ref
```

Comment:
It changes the definition of `apply`, `unapply`, `hashCode`, and `equals` functions implicitly defined by `case class`. Please define `langOpt` as local variables instead of parameters.